plugin/src/functionalTest/kotlin/com/exactpro/th2/gradle/Th2BaseGradlePluginFunctionalTest.kt

```diff
@@ -99,16 +101,149 @@ class Th2BaseGradlePluginFunctionalTest {
+                      "bundles": [
+                        {
+                          "bundleName": "test-license",
+                          "licenseName": "Test-License",
+                        }
+                      ],
+                      "transformationRules": [
+                        {
+                          "bundleName": "test-license",
+                          "licenseNamePattern": ".*"
+                        }
+                      ]
+                    }
+                    """.trimIndent(),
+                )
+            }
+        propertiesFile.writeText(
+            """
+            th2.licence.allow-licence-url=file:${customAllowedLicense.absolutePath}
+            th2.licence.license-normalizer-bundle-path=${customLicenseNormalizer.absolutePath}
+            """.trimIndent(),
+        )
+        val result =
+            GradleRunner.create()
+                .forwardOutput()
+                .withDebug(true)
+                .withConfiguredVersion()
+                .withPluginClasspath()
+                .withProjectDir(projectDir)
+                .withArguments(
+                    "--stacktrace",
+                    "checkLicense",
+                    // because no git repository exist in test
+                    "-x",
+                    "generateGitProperties",
+                )
+                .build()
+
+        val checkLicenses = result.task(":checkLicense")
+        assertNotNull(checkLicenses, "task checkLicense was not executed") {
+            assertEquals(TaskOutcome.SUCCESS, it.outcome, "unexpected task result")
+        }
+
+        val licenses = MAPPER.readTree(projectDir.resolve("build/reports/dependency-license/licenses.json"))
+        assertIs<ObjectNode>(licenses, "incorrect licenses file structure")
+        val dependencies = assertIs<ArrayNode>(licenses.get("dependencies"), "dependencies must be a collection")
+
+        assertAll(
+            {
+                dependencies.assertModuleLicense("io.ktor:ktor-bom", "Test-License")
+            },
+            {
+                dependencies.assertModuleLicense("io.ktor:ktor-server-default-headers", "Test-License")
+            },
+        )
+    }
+
     private fun ArrayNode.assertHasModuleLicenses(moduleName: String) {
         val module =
             elements().asSequence()
                 .find { it.get("moduleName").textValue() == moduleName }
         assertNotNull(module, "module $moduleName not found") { m ->
-            val moduleInfo = assertIs<ObjectNode>(m, "module info muse be an object")
+            val moduleInfo = assertIs<ObjectNode>(m, "module info must be an object")
             assertNotNull(moduleInfo.get("moduleLicenses"), "module $moduleName does not have licenses") {
                 assertIs<ArrayNode>(it, "module $moduleName licenses must be a collection")
                 assertFalse(it.isEmpty, "module $moduleName has empty licenses collection")
             }
         }
     }
```

Comment:
suggestion: I think you can modify this method to avoid having duplicated code
```kotlin
    private fun ArrayNode.assertHasModuleLicenses(moduleName: String, extraAssertion: (ArrayNode) -> Unit = {}) {
        val module =
            elements().asSequence()
                .find { it.get("moduleName").textValue() == moduleName }
        assertNotNull(module, "module $moduleName not found") { m ->
            val moduleInfo = assertIs<ObjectNode>(m, "module info must be an object")
            assertNotNull(moduleInfo.get("moduleLicenses"), "module $moduleName does not have licenses") {
                assertIs<ArrayNode>(it, "module $moduleName licenses must be a collection")
                assertFalse(it.isEmpty, "module $moduleName has empty licenses collection")
                extraAssertion(it)
            }
        }
    }
```